crates/subspace-core-primitives/src/pieces.rs

```diff
@@ -206,6 +206,14 @@ impl PieceIndex {
         self.0.to_le_bytes()
     }
 
+    /// Convert segment index and piece position to piece index.
+    //
+    // TODO: tests
+    #[inline]
+    pub fn from_segment_and_position(segment_index: SegmentIndex, position: u32) -> Self {
```

Comment:
We have more ergonomic APIs than this, I don't think we need it

Reply:
I couldn’t find a more ergonomic API, can you suggest one?

Reply:
See `segment_piece_indexes_source_first` in the comment above